plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend

```diff
@@ -160,10 +163,19 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 
 			if (element instanceof SequenceMessageType || element instanceof ArmTimerMessage) {
 				if (!names.add(getMessagesMapKey(element)) && element.equals(message)) {
-						error('Duplicated message! The same message is already defined.',
+					if (element instanceof SequenceMessageType) {
+						var source = (element as SequenceMessageType).source
+						var target = (element as SequenceMessageType).target
+						error('The same exchange is already used in text editor between \"' +
+							source + "\" and \"" + target + "\"!",
+							TextualScenarioPackage.Literals.MESSAGE__NAME)
+					} else if(element instanceof ArmTimerMessage) {
+						error('The same exchange is already used in text editor on timeline \"' +
+							(element as ArmTimerMessage).name +"\"!",

```

Comment:
shouldn't be here participant instead of name? :-?

Reply:
Yes.